libkrb5/src/context.rs

```diff
@@ -528,16 +525,17 @@ impl Krb5Context {
     }
 
     fn parse_wrap_token(encoded_data: &[u8], usage: Krb5KeyUsage, seq_num: i32) -> Result<(Vec<u8>, Vec<u8>), Krb5Error> {
-        let (mut header, cipher_text) = (encoded_data[..16].to_vec(), &encoded_data[16..]);
-        let rrc = Krb5Context::parse_and_verify_wrap_token_header(header.as_slice(), usage, seq_num).unwrap();
+        let (header, cipher_text) = (encoded_data[..16].to_vec(), &encoded_data[16..]);
+        let rrc = Krb5Context::parse_and_verify_wrap_token_header(header.as_slice(), usage, seq_num)?;
         let cipher_text = Krb5Context::rotate_left(cipher_text, rrc);
 
         Ok((cipher_text, header))
     }
 
     fn parse_and_verify_wrap_token_header(header: &[u8], usage: Krb5KeyUsage, expected_seq_num: i32) -> Result<u16, Krb5Error> {
         let mut parse_wrap_token_header = tuple::<_, _, (&[u8], ErrorKind), _>((be_u16, be_u8, take(1u8), be_u16, be_u16, be_u64));
-        let (_, (token_id, flags, filler, ec, rrc, seq_num)) = parse_wrap_token_header(header).unwrap();
+        let (_, (token_id, flags, filler, _ec, rrc, seq_num)) = parse_wrap_token_header(header).map_err(
```

Comment:
#question
Wouldn't change [1] `LibraryError` to `InvalidToken` if we replaced `map_err` to `?`?

[1]: https://github.com/OneIdentity/libkrb5-rs/pull/2/files#diff-c133b1820114999d94b93a6571c1596a3be6beb96b5d0d73d037b5349a9d35e7L45

Reply:
I tried it, and it would not, but this would:
```rust
impl From<nom::Err<(&[u8], nom::error::ErrorKind)>> for Krb5Error {
    fn from(error: nom::Err<(&[u8], nom::error::ErrorKind)>) -> Self {
        Krb5Error::InvalidToken
    }
}
```
Should I change the map error to this?

Reply:
yes